default/views.py

```diff
@@ -217,7 +224,31 @@ def uploadfile(request):
 
     return JsonResponse({
         'files': [{
+            'id': file.id,
             'name': request_file.name,
             'size': request_file.size,
         }],
     })
+
+
+@require_GET
+def deletefile(request, id):
+    if 'files' not in request.session:
+        return JsonResponse({
+            'message': _('No files to delete')
+        }, status=404)
+
+    to_delete = list(filter(lambda x: x['id'] == str(id), request.session['files']))
+
+    for item in to_delete:
+        request.session['files'].remove(item)
+    request.session.modified = True
+
+    file_found = len(to_delete) > 0
+    response = JsonResponse({
+        'message': _('File deleted') if file_found else _('File not found')
+    })
+
+    if not file_found:
+        response.status_code = 404
```

Comment:
I find having a filter and then a loop a little less straightforward than it could be. The following seems easier to read:

```python
found = False
for fileinfo in request.session['files']:
    if fileinfo['id'] == str(id):
      request.session['files'].remove(item)
      found = True
```

I also think it'd be easier to read to put all 'file found' logic together, e.g.:

```python
if found:
    message = _('File deleted')
else:
    message = _('File not found')
    response.status_code = 404

response = JsonResponse({'message': message})
```

I'm assuming that the session can have multiple files with the same ID? Otherwise, if there can only be one, we can do:

```python
for fileinfo in request.session['files']:
    if fileinfo['id'] == str(id):
        request.session['files'].remove(item)
        message = _('File deleted')
        break
else:
    message = _('File not found')
    response.status_code = 404
```